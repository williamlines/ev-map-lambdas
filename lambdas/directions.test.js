const { directions } = require("./directions");
const mock = require("./__mocks__/directionsMock.json");
const fetchmock = require("jest-fetch-mock");

fetchmock.enableMocks();

describe("direcions", () => {
  it("should not throw", async () => {
    fetch.mockResponseOnce(JSON.stringify(mock));
    const ret = await directions();
    expect(ret).toMatchSnapshot();
  });
});
