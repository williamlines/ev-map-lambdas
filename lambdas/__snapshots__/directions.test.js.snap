// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`direcions should not throw 1`] = `
{
  "body": "OK",
  "message": {
    "code": "Ok",
    "routes": [
      {
        "distance": 4691.4,
        "duration": 1504.1,
        "geometry": {
          "coordinates": [
            [
              -84.518509,
              39.134135,
            ],
            [
              -84.518432,
              39.133835,
            ],
            [
              -84.51906,
              39.133601,
            ],
            [
              -84.519951,
              39.133702,
            ],
            [
              -84.519824,
              39.135164,
            ],
            [
              -84.520051,
              39.135226,
            ],
            [
              -84.520733,
              39.128594,
            ],
            [
              -84.521212,
              39.127956,
            ],
            [
              -84.521548,
              39.124838,
            ],
            [
              -84.518858,
              39.124686,
            ],
            [
              -84.519104,
              39.123542,
            ],
            [
              -84.51722,
              39.12344,
            ],
            [
              -84.517774,
              39.122799,
            ],
            [
              -84.518112,
              39.121535,
            ],
            [
              -84.517362,
              39.120597,
            ],
            [
              -84.518206,
              39.119099,
            ],
            [
              -84.513743,
              39.115317,
            ],
            [
              -84.514554,
              39.114744,
            ],
            [
              -84.514307,
              39.114531,
            ],
            [
              -84.514551,
              39.114249,
            ],
            [
              -84.511692,
              39.102682,
            ],
            [
              -84.511987,
              39.102638,
            ],
          ],
          "type": "LineString",
        },
        "legs": [
          {
            "distance": 4691.4,
            "duration": 1504.1,
            "steps": [],
            "summary": "",
            "weight": 1746.9,
          },
        ],
        "weight": 1746.9,
        "weight_name": "cyclability",
      },
    ],
    "uuid": "qMHLbFTLzijrShRCUTQnGJ1k7R0Is-3Wh4yRz9vaTMMIwvT2fnQvng==",
    "waypoints": [
      {
        "distance": 18.839114182243243,
        "location": [
          -84.518509,
          39.134135,
        ],
        "name": "",
      },
      {
        "distance": 15.959786826422654,
        "location": [
          -84.511987,
          39.102638,
        ],
        "name": "East 6th Street",
      },
    ],
  },
  "statusCode": 200,
}
`;
